include/nanobind/nb_eval.h

```diff
@@ -0,0 +1,69 @@
+/*
+    nanobind/nb_eval.h: Support for evaluating Python expressions and statements
+                        from strings and files
+
+    Adapted by Nico Schlömer from pybind11's eval.h with
+
+    Copyright (c) 2016 Klemens Morgenstern <klemens.morgenstern@ed-chemnitz.de> and
+                       Wenzel Jakob <wenzel.jakob@epfl.ch>
+
+    All rights reserved. Use of this source code is governed by a
+    BSD-style license that can be found in the LICENSE file.
+*/
+
+#pragma once
+
+#include <nanobind/nanobind.h>
+
+#include <utility>
+
+NAMESPACE_BEGIN(NB_NAMESPACE)
+
+enum eval_mode {
+    // Evaluate a string containing an isolated expression
+    eval_expr = Py_eval_input,
+
+    // Evaluate a string containing a single statement. Returns \c none
+    eval_single_statement = Py_single_input,
+
+    // Evaluate a string containing a sequence of statement. Returns \c none
+    eval_statements = Py_file_input
+};
+
+template <eval_mode start = eval_expr>
+object eval(const str &expr, object global = object(), object local = object()) {
```

Comment:
Please change the interface so that `global` and `local` are accepted as `handle`. The caller already holds a reference, so taking objects as input is overkill.

Reply:
I think it used to be `object` so you can pass in `None`. Will change to `handle`.